http-server/src/main/java/io/micronaut/http/server/exceptions/response/DefaultHtmlErrorResponseBodyProvider.java

```diff
@@ -0,0 +1,190 @@
+                                  }            
+                                  b, strong {
+                                    font-weight: 700;
+                                  }            
+                                  i, em {
+                                    font-style: italic;
+                                  }            
+                                  main {
+                                    display: grid;
+                                    gap: 1em;
+                                    padding: 2em;
+                                    place-items: center;
+                                    text-align: center;
+                                  }            
+                                  main header {
+                                    width: min(100%, 18em);
+                                  }            
+                                  main header svg {
+                                    height: auto;
+                                    max-width: 100%;
+                                    width: 100%;
+                                  }            
+                                  main article {
+                                    margin-top: -0.95em;
+                                    width: min(100%, 30em);
+                                  }            
+                                  main article p {
+                                    font-size: 75%;
+                                  }            
+                                  main article br {            
+                                    display: none;            
+                                    @media(min-width: 48em) {
+                                      display: inline;
+                                    }            
+                                  }          
+            """;
+
+    private final MessageSource messageSource;
+    private final LocaleResolver<HttpRequest<?>> localeResolver;
+    private final Map<LocaleStatus, String> cache = new ConcurrentHashMap<>();
+
+    DefaultHtmlErrorResponseBodyProvider(MessageSource messageSource,
+                                         LocaleResolver<HttpRequest<?>> localeResolver) {
+        this.messageSource = messageSource;
+        this.localeResolver = localeResolver;
+    }
+
+    @Override
+    public String body(@NonNull ErrorContext errorContext, @NonNull HttpResponse<?> response) {
+        final HttpStatus status = response.status();
+        Locale locale = localeResolver.resolveOrDefault(errorContext.getRequest());
+        return cache.computeIfAbsent(new LocaleStatus(locale, status), key -> html(locale, status, errorContext));
+    }
+
+    private String html(Locale locale, HttpStatus status, ErrorContext errorContext) {
+        final String errorTitleCode = status.getCode() + ".error.title";
+        final String errorTitle = messageSource.getMessage(errorTitleCode, status.getReason(), locale);
```

Comment:
XSS potential on status.getReason

Reply:
well not right now, since HttpStatus is an enum, but potentially if you fix the status error thing from above.